analysis_scripts/qc_scripts/pipeline_qc.md

```diff
@@ -0,0 +1,74 @@
+pipeline\_qc
+================
+Natalie Davidson
+1/20/2021
+
+## Nature News Pipeline QC
+
+This document is for QC checking for each step of the pipeline. Here, we like to visualize and check statistics that should be stable across each article across all years. The data we will be working with are the following:
+
+1.  Pipeline step 1, download data: `./data/scraped_data/downloads/links_crawled_YEAR.json`
+2.  Pipeline step 2, process data into coreNLP readable: `./data/scraped_data/coreNLP_input_YEAR/*`
+3.  Pipeline step 3, coreNLP output: `./data/scraped_data/coreNLP_output_YEAR/*`
+4.  Pipeline step 4, processed coreNLP output: `./data/quote_table_raw_*.tsv` and `./data/location_table_raw_*.tsv`
+
+**All analysis shown below depends on the functions described in `/analysis_scripts/analysis_utils.R`**
+
+## QC pipeline step 1: Check scrapy output
+
+Make sure there are no empty files. Count the number of files to make sure this is constant across all steps. Count the number of words in each article to compare down the entire pipeline.
+
+#### plotting year level stats
+
+Here we want to plot year level statistics to identify if there was a problem processing any of the years, or to identify trends in the data <img src="pipeline_qc_files/figure-markdown_github/unnamed-chunk-2-1.png" style="display: block; margin: auto;" /><img src="pipeline_qc_files/figure-markdown_github/unnamed-chunk-2-2.png" style="display: block; margin: auto;" /><img src="pipeline_qc_files/figure-markdown_github/unnamed-chunk-2-3.png" style="display: block; margin: auto;" />
+
+#### plotting article level stats
+
+Now lets do a quick plot of article specific stats. So far in the pipeline, we only have the body of the text, so lets just plot that and make sure that it is consistent over time and there are no outliers.
+
+<img src="pipeline_qc_files/figure-markdown_github/unnamed-chunk-3-1.png" style="display: block; margin: auto;" /><img src="pipeline_qc_files/figure-markdown_github/unnamed-chunk-3-2.png" style="display: block; margin: auto;" />
```

Comment:
Just curious why are there 6 duplicated articles for 2020? Was the pipeline ran twice or something?

Reply:
ah, this was caused by a re-direct when accessing a page via the scraper. The 6 documents that are the same are all empty files called "index.html"
When I run the scraper again for 2020 and 2021, I believe they should go away. If not, I will dig deeper. I opened an issue (https://github.com/greenelab/nature_news_disparities/issues/6) so that this is not lost.